test/Result/Result.Tests.cs

```diff
@@ -29,5 +29,35 @@ public void Failure()
             Assert.False(result2);
             Assert.True(result2.IsError);
         }
+
+        [Fact]
+        public async Task SuccessWithTask()
+        {
+            // Arrange
+            Result result1 = Result.Ok();
+            Task<Result> taskResult() => Task.FromResult(result1);
+
+            // Act
+            var result2 = await taskResult();
+
+            //Assert
+            Assert.True(result2);
+            Assert.False(result2.IsError);
+        }
+
+        [Fact]
+        public async Task FailureWithTask()
```

Comment:
Since we already tested the Ok and Fail methods in the first two tests this one is not necessary.